gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java

```diff
@@ -120,9 +155,75 @@ private void resetMetrics(Map<String, AtomicLong> metrics) {
     }
   }
 
+  /**
+   * Updating the required gcs specific statistics based on httpresponse.
+   *
+   * @param statusCode
+   */
+  private void updateGcsIOSpecificStatistics(int statusCode) {
+
+    if (statusCode == 429) {
+      incrementRateLimitingCounter();
```

Comment:
nit: Can consider moving this inside the if in line 173. Once we do that we can return from the if statements. Not a big saving, but can reduce the number of if.

If this can get statuscode=200. Also can add another condition of <=200 and return. This will be most frequent operation, I guess.

Reply:
StatusCode 200 is request succeeded, so maybe statusCode < 200 could have a return statement instead ?